### ISSUE TITLE
Empty files on S3 after gzipping with Django 1.11 (s3boto)

### ISSUE THREAD

**Issue by melwas** [jschneier/django-storages#367]
In Django 1.11.x there was a change in collectstatic command ([here](https://docs.djangoproject.com/en/1.11/releases/1.11/#django-contrib-staticfiles)). 

I noticed that after upgrade there is an issue with files uploaded to S3 when GZIP is enabled. Some files (post-processed by `ManifestFilesMixin` and used in its manifest file) are uploaded as empty files.

I found that there is a potential issue with `_compress_content` method in `S3Boto3Storage` class. 

Currently, the [method](https://github.com/jschneier/django-storages/blob/master/storages/backends/s3boto3.py#L386) looks like this:

```python
zbuf = BytesIO()
zfile = GzipFile(mode='wb', compresslevel=6, fileobj=zbuf)
try:
    zfile.write(force_bytes(content.read()))
finally:
    zfile.close()
zbuf.seek(0)
return zbuf
```

When I use the storage and also use `ManifestFilesMixin` from Django the `content.read()` returns an empty bytes object. If I add `content.seek(0)` before everything works as expected.

When I revert to Django 1.10.7 everything also works as expected. When I disable gzipping the files are also uploaded correctly.

**Comment by jschneier** [comment]
A failing test case would be very helpful here. From reading the linked to Django changelog backwards incompat log it reads as though you have an issue on your side although I might be misunderstanding it.

**Comment by melwas** [comment]
@jschneier I added a failing test with explanation :)

**Comment by jmichalicek** [comment]
Just confirming this issue.  I ran into it and tracked it down very shortly after Django 1.11 was released, but got sidetracked and did not get it reported.

**Comment by jschneier** [comment]
Fixed in #373. Will release shortly.
